### PR TITLE
feat: async language support

### PR DESCRIPTION
**Pull request by SimeonGriggs**
Adds support for using documents as language configuration, like [Internationalized Array](https://github.com/sanity-io/sanity-plugin-internationalized-array) and [Document Internationalization](https://github.com/sanity-io/document-internationalization).

### CODE REVIEW

File: src/LanguageFilterStudioContext.tsx [hunk 1]
```diff
@@ -36,13 +46,37 @@ const LanguageFilterStudioContext =
 export function LanguageFilterStudioProvider(
   props: LayoutProps & LanguageFilterStudioContextProps
 ) {
-  const options = useMemo(
-    () => ({
+  const client = useClient({apiVersion: '2023-01-01'})
+
+  // const deferredDocument = useDeferredValue(document)
```
Comment (snorrees) [inline src/LanguageFilterStudioContext.tsx:51]: Delete instead of out-comment imo

File: src/types.ts [hunk 2]
```diff
@@ -8,20 +8,34 @@ export interface LanguageFilterSchema extends ObjectSchemaType {
   options?: LanguageFilterOptions
 }
 
-export interface Language {
-  id: string
+export type Language = {
```
Comment (snorrees) [inline src/types.ts:11]: Why changed to type?

**Review by snorrees — APPROVED** [review]
Fantastic! Minor nitpicks, proceed as you will :)

File: src/LanguageFilterStudioContext.tsx (hunk 1 above)
Comment (SimeonGriggs) [reply src/LanguageFilterStudioContext.tsx:51]: Good call, removed.

File: src/types.ts (hunk 2 above)
Comment (SimeonGriggs) [reply src/types.ts:11]: Copied over the updated type from Internationalized Array (another reason to have a single @sanity/languages plugin)

**Pull request merged**